eslint.config.mjs

```diff
@@ -38,23 +40,14 @@ module.exports = tseslint.config(
 			tseslint.configs.strictTypeChecked,
 			tseslint.configs.stylisticTypeChecked,
 		],
-		files: ["**/*.js", "**/*.ts"],
+		files: ["**/*.{js,mjs,ts}"],
 		languageOptions: {
 			parserOptions: {
-				projectService: {
-					allowDefaultProject: [
-						"*.config.*s",
-						".eslint-doc-generatorrc.js",
-						"bin/*.js",
-						"src/tests/*.js",
-					],
-				},
-				tsconfigRootDir: __dirname,
+				projectService: { allowDefaultProject: ["*.config.*s"] },
+				tsconfigRootDir: import.meta.dirname,
```

Comment:
This has been causing the `eslint-plugin-n` updates to fail in our other repos, because their supported node version rule flags `import.meta.dirname` for not being available before node v20, and we have v18 in our engines.  Obviously, `engines` is more of a declaration of what consumers of the package should use, and not necessarily apply to repo configs, but just calling out the impact this change has had.   We may want to exclude config files from that rule.  Thoughts?

Ex: https://github.com/JoshuaKGoldberg/eslint-fix-utils/pull/87

Reply:
Hmm, yeah, good point... I've also taken to adding `allowExperimental`: https://github.com/JoshuaKGoldberg/TypeStat/blob/b4820413114ddb2094e9fcfbed71d2f9dec16512/eslint.config.js#L85. Since the APIs have long since stabilized I don't this is likely to miss any real user-facing issues.